### PR TITLE
Fix nested features

### PR DESCRIPTION
**Pull request by oandregal**
This fixes the algorithm that merges core, theme, and user preferences for `features`.

## Test

- Use a theme that declares support for link color but nothing else via theme.json. Test case at https://github.com/nosolosw/global-styles-theme/pull/17
- Make sure custom colors are still enabled.
- Make sure the link color is enabled.

Before this PR custom colors won't be enabled.

### CODE REVIEW

File: lib/global-styles.php [hunk 1]
```diff
@@ -521,11 +521,16 @@ function gutenberg_experimental_global_styles_merge_trees( $core, $theme, $user
...
-			$result[ $block_name ][ $subtree ] = array_merge(
-				$core[ $block_name ][ $subtree ],
-				$theme[ $block_name ][ $subtree ],
-				$user[ $block_name ][ $subtree ]
+		$result[ $block_name ][ 'presets' ] = array_merge(
+			$core[ $block_name ][ 'presets' ],
+			$theme[ $block_name ][ 'presets' ],
+			$user[ $block_name ][ 'presets' ]
+		);
+		foreach ( array_keys( $core[ $block_name ]['features'] ) as $subtree ) {
```
Comment (youknowriad) [inline lib/global-styles.php:529]: The thing is here, I'm not sure we will only have arrays here, maybe some keys in "features" will contain scalar values in the end.

Comment (nosolosw) [reply lib/global-styles.php:529]: Ah, right. This won't work _if_ we do that.

Given that we don't have that use case yet, that we're coming with a big refactoring of theme.json, and that the `colors` is now broken, would you think we can land this PR as it is and iterate?

Comment (youknowriad) [reply lib/global-styles.php:529]: Yes, I'm ok with that.

**Review by youknowriad — APPROVED** [review]

**Comment by nosolosw** [comment]
There's a CI error as a result of package-lock not being updated. I'll rebase and merge after https://github.com/WordPress/gutenberg/pull/25223 lands

**Pull request merged**